force-app/main/default/classes/WeatherService.cls

```diff
@@ -0,0 +1,33 @@
+global class WeatherService {
+    @AuraEnabled(cacheable=true)
+    public static WeatherData getCurrentWeather(String location){ 
+           
+        WeatherAPIKey__mdt weaherApiKey = [SELECT Key__c FROM WeatherAPIKey__mdt LIMIT 1];  
+        String apiKey = weaherApiKey.Key__c;
```

Comment:
check if weaherApiKey exists first and after that take value from property, in this particular case Key__c